lighthouse-core/audits/first-meaningful-paint.js

```diff
@@ -83,9 +74,23 @@ class FirstMeaningfulPaint extends Audit {
     });
   }
 
-  static calculateScore(data) {
-    // First meaningful paint is the last timestamp observed from the candidates
-    const firstMeaningfulPaint = (data.fMP.ts - data.navStart.ts) / 1000;
+  static calculateScore(evts) {
+    const firstMeaningfulPaint = (evts.firstMeaningfulPaint.ts - evts.navigationStart.ts) / 1000;
+    const firstContentfulPaint = (evts.firstContentfulPaint.ts - evts.navigationStart.ts) / 1000;
```

Comment:
it should handle that. The audit is only looking at the first fCP that has the same frameID as the first tracingStartedInPage. 

The [traces in fixtures](https://cloud.githubusercontent.com/assets/39191/21165321/7a1452d6-c153-11e6-8822-7ac58fea4c94.png) have multiple fCP's per trace already. :)